src/Event/Http/FpmHandler.php

```diff
@@ -280,17 +278,8 @@ private function waitUntilStopped(int $pid): void
     /**
      * Return an array of the response headers.
      */
-    private function getResponseHeaders(ProvidesResponseData $response, bool $isMultiHeader): array
+    private function getResponseHeaders(ProvidesResponseData $response): array
     {
-        $responseHeaders = $response->getHeaders();
-        if (! $isMultiHeader) {
-            // If we are not in "multi-header" mode, we must keep the last value only
-            // and cast it to string
-            foreach ($responseHeaders as $key => $value) {
-                $responseHeaders[$key] = end($value);
-            }
-        }
-
-        return array_change_key_case($responseHeaders, CASE_LOWER);
```

Comment:
Headers are no returned unchanged because we normalize them later in any case. That simplifies the code.